3545/2/commands/checkin.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module CheckInCommand
+  TOKEN = '871720433:AAE0re0dH98iwJ98bBgL-14YaB8ZjvVdk9s'
+  API_URL = "https://api.telegram.org/bot#{TOKEN}/getFile?file_id="
+  FILE_API_URL = "https://api.telegram.org/file/bot#{TOKEN}/"
+
+  def checkin!
+    if session[:is_checkedin?]
+      respond_with :message, text: 'Вы в кемпе'
+    elsif session[:id]
+      save_context :upload_photo
+      session[:is_checkedin?] = true
+      respond_with :message, text: 'Пришли мне себяшку'
+    else
+      respond_with :message, text: 'Зарегистрируйтесь!'
+    end
+  end
+
+  def upload_photo(*)
+    create_checkin_dir
+
+    File.open(checkin_path + '/selfie.jpg', 'wb') do |file|
+      file << URI.open(FILE_API_URL + image_path).read
+    end
+
+    save_context :upload_geo
+    respond_with :message, text: 'Точно пришёл на место?'
+  end
+
+  def upload_geo(*)
+    File.open(checkin_path(chat['id']) + '/geo.txt', 'wb') do |file|
+      p payload
+      file << payload['location'].to_s
+    end
+    respond_with :message, text: 'Отлично, порви сегодня всех. За себя и за Сашку.'
+  end
+
+  def create_checkin_dir
+    FileUtils.mkdir_p(checkin_path)
```

Comment:
are you creating each time? maybe it's better to check if it exists?

Reply:
Sure, good point

Reply:
why it's not fixed here?